File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/validation/validator/UniqueCollectCategoryValidator.java
package com.windea.demo.cloudcollect.core.validation.validator;

import com.windea.demo.cloudcollect.core.domain.entity.CollectCategory;
import com.windea.demo.cloudcollect.core.service.CollectCategoryService;
import com.windea.demo.cloudcollect.core.validation.annotation.UniqueCollectCategory;
import org.springframework.beans.factory.annotation.Autowired;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

/**
 * 唯一收藏分类的校验器。
 */
public class UniqueCollectCategoryValidator implements ConstraintValidator<UniqueCollectCategory, CollectCategory> {
	@Autowired
	private CollectCategoryService service;

	@Override
	public boolean isValid(CollectCategory value, ConstraintValidatorContext context) {
		return !service.exists(value);
	}
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/validation/validator/UniqueCollectCategoryValidator.kt
package com.windea.demo.cloudcollect.core.validation.validator

import com.windea.demo.cloudcollect.core.domain.entity.*
import com.windea.demo.cloudcollect.core.repository.*
import com.windea.demo.cloudcollect.core.validation.annotation.*
import org.springframework.beans.factory.annotation.*
import javax.validation.*

/**唯一收藏分类的校验器。*/
open class UniqueCollectCategoryValidator : ConstraintValidator<UniqueCollectCategory, CollectCategory> {
	@Autowired private lateinit var categoryRepository: CollectCategoryRepository
	
	override fun isValid(value: CollectCategory, context: ConstraintValidatorContext): Boolean {
		return !categoryRepository.existsByNameAndUser(value.name, value.user)
	}
}
